sync/block_listener.go

```diff
@@ -91,9 +91,32 @@ func (bl *BlockListener) handleBlock(data service.GossipMessage) {
 	// set the block id when received
 	blk.Initialize()
 
-	bl.Log.With().Info("got new block", blk.Fields()...)
+	activeSet := 0
+	if blk.ActiveSet != nil {
+		activeSet = len(*blk.ActiveSet)
+	}
+
+	refBlock := ""
+	if blk.RefBlock != nil {
+		refBlock = blk.RefBlock.String()
+	}
+	bl.Log.With().Info("got new block",
+		blk.ID(),
+		blk.LayerIndex,
+		blk.LayerIndex.GetEpoch(),
+		log.String("sender_id", blk.MinerID().ShortString()),
+		log.Int("tx_count", len(blk.TxIDs)),
+		//log.Int("atx_count", len(blk.ATXIDs)),
+		log.Int("view_edges", len(blk.ViewEdges)),
+		log.Int("vote_count", len(blk.BlockVotes)),
+		blk.ATXID,
+		log.Uint32("eligibility_counter", blk.EligibilityProof.J),
+		log.String("ref_block", refBlock),
+		log.Int("active_set", activeSet),
+	)
 	// check if known
 	if _, err := bl.GetBlock(blk.ID()); err == nil {
+		data.ReportValidation(config.NewBlockProtocol)
```

Comment:
this should be reconsidered or changed a bit imo, if we get a block that we already know we will just propagate it again. gossip should prevent it but here we validate that it's ok and even without syntactically checking the block. this should probably be a special case for only when this is our block ?

Reply:
if this is in our DB it is syntactically valid, do you still think we should have another check?

Reply:
again if this is our block then ok. otherwise I need to think if a block can somehow bypass gossip and get into this codepath again after it's in the db.